File: Problems 0400-0499/424. Longest Repeating Character Replacement/solution.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] freq = new int[26];
        int currMaxFreq = 0;
        int left = 0;
        int res = 0;
        for (int right = 0; right < s.length(); right++) {
            char c = s.charAt(right);
            freq[c - 'A']++;
            
            if (currMaxFreq < freq[c - 'A']) {
                currMaxFreq = freq[c - 'A'];
            }

            if (right - left + 1 - currMaxFreq > k) {
                freq[s.charAt(left++) - 'A']--;
            }

            res = right - left + 1;
        }
        return res;
    }
}

File: Problems 0400-0499/424. Longest Repeating Character Replacement/solution.ts
function characterReplacement(s: string, k: number): number {
    const freq = Array(26).fill(0);

    let left = 0;
    let currMaxCount = 0;
    let res = 0;
    for (let right = 0; right < s.length; right++) {
        const charCode = s[right].charCodeAt(0) - 'A'.charCodeAt(0);
        freq[charCode]++;

        if (currMaxCount < freq[charCode]) {
            currMaxCount = freq[charCode];
        }

        if (right - left + 1 - currMaxCount > k) {
            freq[s[left].charCodeAt(0) - 'A'.charCodeAt(0)]--;
            left++;
        }

        res = Math.max(res, right - left + 1);
    }

    return res;
};